lib/marriage_abroad_outcome_flattener.rb

```diff
@@ -0,0 +1,90 @@
+class MarriageAbroadOutcomeFlattener
+  COUNTRIES_DIR = "lib/smart_answer_flows/marriage-abroad/outcomes/countries".freeze
+
+  FEE_TABLE_PARTIAL = <<~FEE_TABLE.freeze
+    <%= render partial: 'consular_fees_table_items.govspeak.erb',
+        collection: calculator.services,
+        as: :service,
+        locals: { calculator: calculator } %>
+  FEE_TABLE
+
+  HOW_TO_PAY_PARTIAL = <<~HOW_TO_PAY.freeze
+    <%= render partial: 'how_to_pay.govspeak.erb', locals: {calculator: calculator} %>
+  HOW_TO_PAY
+
+  TITLE_PARTIAL_CONTENT = <<~TITLE.freeze
+    <% if calculator.partner_is_same_sex? %>
+      Civil partnership in COUNTRY
+    <% else %>
+      Marriage in COUNTRY
+    <% end %>
+  TITLE
+
+  PAYMENT_RE = /^\^?You can( only)? pay by.*?\n.*?^$/mi
+  SERVICE_FEE_RE = /^Service \| Fee\n.*?^$/mi
+
+  attr_reader :country, :logger
+
+  def initialize(country, logger = STDOUT)
+    @country = country
+    @logger = logger
+  end
+
+  def flatten
+    logger.puts "Flattening outcomes for #{country}"
+    remove_old_country_partials
+    copy_test_artefacts
+    rename_and_add_partials_to_country_files
+    create_title_partial
+  end
+
+private
+
+  def rename_and_add_partials_to_country_files
+    Dir["#{country_partials_dir}/**/*.txt"].each do |src_filename|
```

Comment:
Are the filenames pretty rigid here? It looks like this would all work good so long as the file was named `same_sex.txt` or `opposite_sex.txt`but looks like it could go a bit wacky (unless I've missed something) if the glob matched something else.